### PR TITLE
Windows CI

### PR DESCRIPTION
**Pull request by kanye-quest**
# Summary
- Uses Github Actions for Windows builds and tests
  - Supports MPI-enabled MFEM with MS-MPI and vcpkg-installed Metis
- Make `tests/unit` a CMake subproject, in order to generate the "all unit tests" target for Windows
- Updates Ubuntu runner version to 20.04
- Removes Appveyor CI

TODO:
- [x] merge mfem/github-actions#6
- [x] change references to fork action after merging

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2973](https://github.com/mfem/mfem/pull/2973) | @kanye-quest | @tzanio | @acfisher + @v-dobrev + @tzanio | 4/29/22 | 5/17/22 | 5/18/22 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by tzanio** [comment]
Note: this should probably be reviewed together with https://github.com/mfem/github-actions/pull/6

File: tests/unit/CMakeLists.txt [hunk 1]
```diff
@@ -9,11 +9,8 @@
 # terms of the BSD-3 license. We welcome feedback and contributions, see file
 # CONTRIBUTING.md for details.
 
-# Include the top mfem source directory - needed by some tests, e.g. to
-# #include "general/text.hpp".
-include_directories(BEFORE ${PROJECT_SOURCE_DIR})
-# Include the build directory where mfem.hpp and mfem-performance.hpp are.
-include_directories(BEFORE ${PROJECT_BINARY_DIR})
```
Comment (tzanio) [inline tests/unit/CMakeLists.txt:16]: Is this really not needed?

File: .github/workflows/mfem-analysis.yml [hunk 2]
```diff
@@ -53,11 +53,13 @@ jobs:
       uses: actions/cache@v2
       with:
         path: ${{ env.HYPRE_TOP_DIR }}
-        key: ${{ runner.os }}-build-${{ env.HYPRE_TOP_DIR }}-v2.0
+        # TODO: update to new tag
+        key: ${{ runner.os }}-build-${{ env.HYPRE_TOP_DIR }}-master2
```
Comment (tzanio) [inline .github/workflows/mfem-analysis.yml:57]: This should probably be `-v2.2` or maybe `-v3.0`

**Review by tzanio — APPROVED** [review]
Thanks so much for doing this @kanye-quest 🙏

File: tests/unit/CMakeLists.txt (hunk 1 above)
Comment (kanye-quest) [reply tests/unit/CMakeLists.txt:16]: I added a `target_include_directories` statement for the mfem library target, so the correct includes should be passed to any cmake targets that depend on mfem.

**Comment by tzanio** [comment]
The parallel debug windows test is failing 👇

**Comment by tzanio** [comment]
@acfisher and @v-dobrev, can you please review when you get a chance?

**Review by acfisher — APPROVED** [review]

File: .github/workflows/builds-and-tests.yml [hunk 3]
```diff
@@ -142,6 +143,11 @@ jobs:
       if: matrix.mpi == 'par' && matrix.os == 'windows-2022'
       uses: mpi4py/setup-mpi@v1.0.3
 
+    - name: setup Windows-specific environment vars
+      if: matrix.os == 'windows-2022'
+      run: |
+        echo 'HYPRE_BUILD_SYSTEM=cmake' >> ${Env:GITHUB_ENV}
+
```
Comment (adrienbernede) [inline .github/workflows/builds-and-tests.yml:150]: I would have preferred the duplication of `get hypre` into `get hypre (Windows)` and `get hypre`, to be consistent with the rest.

**Review by adrienbernede — APPROVED** [review]
Looks good to me. Maybe a last change for consistency?

**Comment by v-dobrev** [comment]
I think we should keep the Apveyor CI running for some time along this new windows CI. We can disable it later.

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**